azkaban-common/src/main/java/azkaban/executor/ExecutorManagerAdapter.java

```diff
@@ -88,6 +88,18 @@ public ExecutableFlow getExecutableFlow(int execId)
 
   public List<ExecutableFlow> getRunningFlows() throws IOException;
 
+  /**
+   * <pre>
+   * Returns All running with executors and queued flows
+   * Note, returns empty list if there isn't any running or queued flows
+   * </pre>
+   *
+   * @return
+   * @throws IOException
+   */
+  public List<Pair<ExecutableFlow, Executor>> getActiveFlowsWithExecutor()
```

Comment:
Even I want a separate class, Ideally I will prefer to recycle executionReference but We need to be backward compatible and Pair<ExecutableFlow, Executor> is used through out ExecutorManager. So we cannot make this change without an interface change.